tidb: add `general-log-file` and `compression` (#16663)
pingcap/docs: tidb-configuration-file.md

- Default value: `10000`
- When the number of query rows (including the intermediate results based on statistics) is larger than this value, it is an `expensive` operation and outputs log with the `[EXPENSIVE_QUERY]` prefix.

### `general-log-file` <span class="version-mark">New in v8.0.0</span>

> + The file name of the [general log](/system-variables.md#tidb_general_log).

Reviewer: ```suggestion
+ The filename of the [general log](/system-variables.md#tidb_general_log).
```
https://developers.google.com/style/word-list#filename

> + Default value: ""

Reviewer: ```suggestion
+ Default value: `""`
```

> + If you set file name, the general log is output to this file. If the value is empty, the general log is output to the file specified by [`filename`](#filename) on the TiDB server.

Author: ```suggestion
+ If you specify a file name, the general log is output to this file. If the value is empty, the general log is output to the file specified by [`filename`](#filename) on the TiDB server.
```

> + If you specify a file name, the general log is output to this file. If the value is empty, the general log is output to the file specified by [`filename`](#filename) on the TiDB server.

Reviewer 2: ```suggestion
+ If you specify a file name, the general log is written to this particular file. If the value is blank, the general log is written to the server log of the TiDB instance. The name of server log is specified by [`filename`](#filename) . 
```

> + If you specify a file name, the general log is written to this particular file. If the value is blank, the general log is written to the server log of the TiDB instance. The name of server log is specified by [`filename`](#filename). 

Reviewer: ```suggestion
+ If you specify a filename, the general log is written to this specified file. If the value is blank, the general log is written to the server log of the TiDB instance. You can specify the name of the server log using [`filename`](#filename).
```

- Default value: `0`
- All the log files are retained by default. If you set it to `7`, seven log files are retained at maximum.

> #### `compress` <span class="version-mark">New in v8.0.0</span>

Reviewer 3: ```suggestion
#### `compression` <span class="version-mark">New in v8.0.0</span>
```

+ The compression method for the log.

> + Default value: ""

Reviewer: ```suggestion
+ Default value: `""`
+ Value options: `""`, `"gzip"`
```

> + By default, compression is not enabled. Currently, the only supported compression format is `"gzip"`. After compression is enabled, all log files are affected, such as `slow-log-file` and `general-log-file`.

Reviewer: ```suggestion
+ The default value is `""`, which means no compression. To enable the gzip compression, set this value to `"gzip"`. After compression is enabled, all log files are affected, such as [`slow-query-file`](#slow-query-file) and [`general-log-file`](#general-log-file).
```
